issue_processing/issue_processing.py

```diff
@@ -131,14 +269,83 @@ def reformat(issue_data):
                                not (event["user"] is None or event["ref_target"] is None)]
 
+        # sorts eventsList by time
+        issue["eventsList"] = sorted(issue["eventsList"], key=lambda k: k["created_at"])
+
+    return issue_data
+
+
+def reformat_events(issue_data):
+    """
+    Re-format event information dependent on the event type.
+
+    :param issue_data: the data of all issues that shall be re-formatted
+    :return: the issue data with updated event information
+    """
+
+    log.info("Update event information ...")
+
+    for issue in issue_data:
+
+        # re-format information of every event in the eventsList of an issue
+        for event in issue["eventsList"]:
+
+            if event["event"] == "closed":
+                event["event"] = "state_updated"
+                event["event_info_1"] = "closed"  # new state
+                event["event_info_2"] = "open"  # old state
+                issue["state_new"] = "closed"
+
+            elif event["event"] == "reopened":
+                event["event"] = "state_updated"
+                event["event_info_1"] = "open"  # new state
+                event["event_info_2"] = "closed"  # old state
+                issue["state_new"] = "reopened"
+
+            elif event["event"] == "labeled":
+                label = event["label"]["name"].lower()
+                event["event_info_1"] = label
+
+                # if the label is in this list, it also is a type of the issue
+                if label in known_types:
+                    issue["type"].append(str(label))
+
+                    # creates an event for type updates and adds it to the eventsList
+                    type_event = dict()
+                    type_event["user"] = issue["user"]
+                    type_event["created_at"] = event["created_at"]
+                    type_event["event"] = "type_updated"
+                    type_event["event_info_1"] = label
+                    type_event["event_info_2"] = ""
+                    type_event["ref_target"] = ""
+                    issue["eventsList"].append(type_event)
+
+                # if the label is in this list, it also is a resolution of the issue
+                elif label in known_resolutions:
+                    issue["resolution"].append(str(label))
+
+                    # creates an event for resolution updates and adds it to the eventsList
+                    resolution_event = dict()
+                    resolution_event["user"] = issue["user"]
+                    resolution_event["created_at"] = event["created_at"]
+                    resolution_event["event"] = "resolution_updated"
+                    resolution_event["event_info_1"] = label
+                    resolution_event["event_info_2"] = ""
+                    resolution_event["ref_target"] = ""
+                    issue["eventsList"].append(resolution_event)
+
+            # %TODO: event for removed labels has to be checked. Data is missing
```

Comment:
There should already be examples for `unlabeled` in the most recent GH data. If not, let us now.